api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -815,6 +815,9 @@ message HttpConnectionManager {
   // Append the `x-forwarded-port` header with the port value client used to connect to Envoy. It
   // will be ignored if the `x-forwarded-port` header has been set by any trusted proxy in front of Envoy.
   bool append_x_forwarded_port = 51;
+
+  // Whether the HCM will add ProxyProtocolFilterState to the Connection lifetime filter state. Defaults to `true`.
```

Comment:
I don't recall the history here, but I think this behavior was added to support ProxyProto on the upstream without ProxyProto on the downstream? I think it'd help to clarify the purpose of HCM inserting the filter state.

Reply:
This was added in https://github.com/envoyproxy/envoy/pull/16598. The purpose seems to be so that upstreams that have proxy protocol enabled will have the correct state.

Reply:
Please add some documentation here about cases where it makes sense to set this to false.